### ISSUE TITLE
Tab order of text in blocks in not consistent

### ISSUE THREAD

**Issue by mscherotter** [rachel-fenichel/blockly-keyboard-experimentation#672]
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Description

Using Windows 11 & Microsoft Edge or Google Chrome
The tab order of text in blocks is not consistent - left to right, top to bottom as can be seen in this video recording using the Narrator tool in Windows. I am just pressing the Tab key to move the cursor:

See video recording:
https://github.com/user-attachments/assets/d8fb3393-0547-4136-a0cc-8ead34bc5bba

### Reproduction steps

1. Navigate to https://google.github.io/blockly-keyboard-experimentation/
2. Place a Loops/Count-with block on the canvas
3. Use the tab key to narrate the block contents
4. Notice that the tab order is not expected: first fields and then static text.  it should be left to right, top to bottom.

### Stack trace

```shell

```

### Browsers

Chrome desktop

**Comment by BenHenning** [comment]
This is actually a really interesting issue--thanks for filing it and especially for the recorded video. I'm surprised to see how tab is behaving with Narrator. Unless I'm mistaken, it actually seems to be violating the spec here. Per https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/tabindex a negative tab index should not be reachable with sequential navigation, and in fact those `text` elements don't even have a negative tab index.

I can't easily test with Navigator, but I will need to look into why it's doing this.

**Comment by mscherotter** [comment]
Please reach out if you need some help testing.

Michael S. Scherotter
***@***.***
________________________________

**Comment by BenHenning** [comment]
So https://github.com/google/blockly/pull/9384 might fix this, or at least a big part of this. I noticed when working on that PR that field labels were being included in the accessibility node tree. What's odd is that they weren't focusable, but I'm also underestimating the fact that there are default tab index behaviors for certain elements that likely include text. I think Narrator is simply navigating the nodes. #9384 removes field labels from the tree entirely which will solve at least that part of the issue.

We also have https://github.com/google/blockly/pull/9304 which will remove other nodes (if any) and ensure the correct hierarchical relationship between them (since right now we're largely linearizing even the tree view). That will probably fix other inconsistencies here.

The main problem, however, is why Narrator is even doing this in the first place when tabbing should only never iterate through tabbable (i.e. tabindex of 0) elements, and otherwise users need to use arrow keys to leverage Blockly's keyboard navigation support (which integrates with a screen reader). That's still a big unknown that needs investigation.

**Comment by BenHenning** [comment]
@mscherotter if you get a chance could you please test this again with https://google.github.io/blockly-keyboard-experimentation/screenreader/ and see if the same problems exist? I'm really hoping that the removal of labels from the node tree resolves most of the problem here, but I'm still assuming there will be other parts that are broken yet since we're really not expecting tab key to do anything but go to elements with a tabindex of 0.

**Comment by BenHenning** [comment]
@rachel-fenichel fortunately has a Windows machine and was able to do some testing today on this issue. Her words:

> I have the test page up. Focus is getting desynced from the Narrator cursor and the result is not very usable

And then:

> Ah, the problems I was seeing came from Narrator's [scan mode](https://support.microsoft.com/en-us/windows/chapter-3-using-scan-mode-7b2af804-5a2f-90fd-b0e0-672f7cbbf2da), which turns on automatically for browsers. I turned that off and it behaves as I expect, although it reads out "Blockly workspace region" more often than I like.

The region behavior is also happening on ChromeVox (or at least some form is), so I suspect that perhaps you were running into scan mode originally @mscherotter or perhaps the underlying issues _have_ been fixed. The scan mode may well cause us issues, but it seems that Narrator is behaving correctly without it.

Closing this for now. Please reopen if I missed something or file a new issue if other problems are observed. Thanks again for filing this!
